docs/hoc/modelspec/programmatic/mechanisms/nmodl2.rst

```diff
@@ -428,27 +428,72 @@ PROTECT
 ~~~~~~~
 
 Description:
-    Mod files that assign values to GLOBAL variables are not considered 
-    thread safe. If the mod file is using the GLOBAL as a counter, prefix 
-    the offending assignment statements with the PROTECT keyword so that 
-    multiple threads do not attempt to update the value at the same time 
-    (race condition). If the mod file is using the GLOBAL essentially as 
-    a file scope LOCAL along with the possibility of passing values back 
-    to hoc in response to calling a PROCEDURE, use the THREADSAFE keyword 
-    in the NEURON block to automatically treat those GLOBAL variables 
-    as thread specific variables. NEURON assigns and evaluates only 
-    the thread 0 version and if FUNCTIONs and PROCEDUREs are called from 
-    Python, the thread 0 version of these globals are used.
+    .. code-block::
+
+        NEURON {
+            GLOBAL var
+        }
+
+        BREAKPOINT {
+            PROTECT var = var + 1
+        }
 
-    ``TODO``: Add existing example mod file (share/demo/release/mcna.mod)
+    Mod files that update values to :ref:`GLOBAL` variables are not considered
+    thread safe. In case of multi-threaded/SIMD/GPU execution, such updates can result
+    in a race condition. To avoid this, one needs to use ``PROTECT`` keyword. Note that
+    ``PROTECT`` internally uses atomic operations on CPU or GPU execution and hence
+    the statement needs to be of a simple form such as:
+
+    .. code-block::
+
+        var1 = var2
+        var1 = var1 binary_operator expression
+        var1 = expression binary_operator var1
```

Comment:
I think these are the only two non-duplicate valid forms out of the 5?

Reply:
👍  ( my bad - I was copying the `structured block` expressions from `capture` clause 😞 )